tests/Unit/Domain/Test_InterfaceHelpers.cpp

```diff
@@ -97,6 +97,14 @@ void test_interface_apply(
       interface_apply<DirectionsTag, ComputeSomethingOnInterface>(box, 2.);
   /// [interface_apply_example_stateless]
   CHECK(computed_numbers_with_struct == computed_number_on_interfaces);
+
```

Comment:
please test the other cases you added

Reply:
I think the test cases now cover all code paths